core/MMU.cpp

```diff
@@ -0,0 +1,57 @@
+#include "MMU.hpp"
+
+namespace olympia {
+    const char MMU::name[] = "mmu";
+
+    void MMU::setTLB(olympia::SimpleTLB &tlb) {
+        tlb_cache_ = &tlb;
+    }
+
+    MMU::MMU(sparta::TreeNode *node, const MMUParameterSet *p) :
+            sparta::Unit(node),
+            tlb_always_hit_(p->tlb_always_hit),
+            mmu_latency_(p->mmu_latency) {
+    }
+
+    bool MMU::Lookup(const MemoryAccessInfoPtr &mem_access_info_ptr) {
+        const InstPtr &inst_ptr = mem_access_info_ptr->getInstPtr();
+        uint64_t vaddr = inst_ptr->getTargetVAddr();
+
+        bool tlb_hit = false;
+
+        // C++ comma operator: assign tlb_hit first, then evaluate it. Just For Fun
+        if (tlb_hit = tlb_always_hit_, tlb_hit) {
+        } else {
```

Comment:
Might as well convert to 
```suggestion
        bool tlb_hit = tlb_always_hit_;
        if(false == tlb_hit) {
             // miss code
```